src/Psl/TCP/TLS/Certificate.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\TCP\TLS;
+
+/**
+ * Represents a TLS certificate for secure TCP connections.
+ *
+ * This class encapsulates the necessary information for a TLS certificate, including
+ * the certificate file itself, an optional private key file, and an optional passphrase
+ * for the key.
+ *
+ * @immutable
+ */
+final class Certificate
+{
+    /**
+     * Constructor for the Certificate class.
+     *
+     * @param non-empty-string $certificateFile The path to the certificate file.
+     * @param non-empty-string|null $keyFile The path to the private key file associated with the certificate, if any.
+     * @param non-empty-string|null $passphrase The passphrase for the private key file, if the file is encrypted.
+     *
+     * @pure
+     */
+    public function __construct(
+        public readonly string  $certificateFile,
+        public readonly ?string $keyFile = null,
+        public readonly ?string $passphrase = null,
```

Comment:
#[\SensitiveParameter]

Reply:
Ah! we actually have many places where we should add the attribute, mainly `Psl\Password` and `Psl\Hash` components.